docs/build-mvm/how-to/schedule_functions.md

```diff
@@ -1 +1,875 @@
+      -- Note that only a **spell** can store the log in its logs storage
+      store_log(log: string) -> UnitValue
+        -- Return all logs ordered by timestamp ascending.
+        get_logs() -> GetLogsResult
+    ```
+    
+- Example
+    
+    Note that the spell that just writes to its logs is the default template created by the Fluence CLI. Here’s another — though almost identical — way to write a simple spell with logging: 
+    
+    ```aqua
+    aqua MyFirstSpell
+    
+    export spell
+    
+    import "@fluencelabs/spell/spell_service.aqua"
+    
+    func spell():
+      Spell "self"
+      Spell.store_log("I was run!")
+    ```
+    
+    You can read the logs using the Fluence CLI command like this:
+    
+    ```bash
+    $ fluence deal logs myDeployment myFirstSpell
+    Connecting to random local relay: /ip4/127.0.0.1/tcp/9991/ws/p2p/12D3KooWFS4WXar3f1SWCykTUy9cVKNU8x1yDA18ZYp86mXTUyAe
+    Connected
+    myDeployment (host_id: 12D3KooWFS4WXar3f1SWCykTUy9cVKNU8x1yDA18ZYp86mXTUyAe, worker_id: 12D3KooWBPfFGP7y1jiKgQ4DEkuS87KsfcPt5DDVoWUiwNtbzYFE, spell_id: b030cf69-7a35-4eeb-bad1-981797e67eec):
+    
+    2024-02-09 15:49:15 I was run!
+    ```
+    
+    Or you can get the logs manually via Aqua. Copy this code into your `src/aqua/main.aqua`:
+    
+    ```aqua
+    aqua Main
+    
+    import "@fluencelabs/aqua-lib/builtin.aqua"
+    import "@fluencelabs/aqua-lib/subnet.aqua"
+    import "@fluencelabs/spell/spell_service.aqua"
+    
+    use "deals.aqua"
+    
+    export testSpell
+    
+    func testSpell() -> []GetLogsResult:
+        deals <- Deals.get()
+        dealId = deals.myDeployment!.dealIdOriginal
+    
+        logs: *GetLogsResult
+        on HOST_PEER_ID:
+            subnet <- Subnet.resolve(dealId)
+            if subnet.success == false:
+                Console.print(["Failed to resolve subnet: ", subnet.error])
+    
+            for w <- subnet.workers:
+                on w.worker_id! via w.host_id:
+                    Spell "myFirstSpell"
+                    logs <<- Spell.get_logs()
+        <- logs
+    ```
+    
+    Then call it via the Fluence CLI:
+    
+    ```bash
+    $ fluence run -f 'testSpell()' -i src/aqua/main.aqua
+    Connecting to random local relay: /ip4/127.0.0.1/tcp/9991/ws/p2p/12D3KooWFS4WXar3f1SWCykTUy9cVKNU8x1yDA18ZYp86mXTUyAe
+    Connected
+    [
+      {
+        "error": "",
+        "logs": [
+          {
+            "message": "I was run!",
+            "timestamp": 1707493755
+          }
+        ],
+        "success": true
+      }
+    ]
+    ```
+    
+
+### Spell Mailbox API
+
+The Spell Service provides the Mailbox API to receive messages from the outside world. Note that the mailbox keeps only 50 last messages in the storage.
+
+- API
+    
+    ```aqua
+    
+    data MailboxMessage:
+      -- How put the message into the mailbox
+      init_peer_id: string
+      -- When the message was put
+      timestamp: u64
+      -- The message
+      message: string
```

Comment:
Would be awesome to store the tetraplet and all other metadata for recorded messages...